apps/nextjs/src/components/ContextProviders/ChatProvider.tsx

```diff
@@ -383,7 +392,37 @@ export function ChatProvider({ id, children }: Readonly<ChatProviderProps>) {
       ? lastModeration
       : toxicInitialModeration;
 
-  const ailaStreamingStatus = useAilaStreamingStatus({ isLoading, messages });
+  const {
+    status: ailaStreamingStatus,
+    streamingSection,
+    streamingSections,
+  } = useAilaStreamingStatus({ isLoading, messages });
+
+  const [streamingSectionCompleted, setStreamingSectionCompleted] = useState<
+    LessonPlanKeys | undefined
+  >(undefined);
+
+  const { sectionRefs, setSectionRef } = useLessonPlanScrollManagement(
+    streamingSection,
+    streamingSectionCompleted,
+  );
+
+  const workingMessage = useRef<Message | undefined>(undefined);
```

Comment:
We have to snapshot the message so that we don't get a render loop

Reply:
These kind of comments would be useful in the code itself so we know why they're there when we next try to tidy it

I actually don't quite understand the render loop issue?